src/Slot-Core/Slot.class.st

```diff
@@ -286,6 +286,12 @@ Slot >> name: aSymbol [
 	name := aSymbol
 ]
 
+{ #category : #accessing }
+Slot >> named: aSymbol [
+	"to be polymorhic with slot class"
+	self name: aSymbol
+]
```

Comment:
I understand the intention and maybe it is not important. But according to other similar methods (like reversed, sorted, with ~ed suffix)  I would expect new instance in that case instead of modification of receiver. 
And I can imagine the use case for this. For example rename refactoring could ask existing slot to produce new one with new name. So the #named: message will return a new copy with given name.

Reply:
yes, I have it in the backlog... with the composition I need to clean this up anyway.